src/core/api/db_push.pl

```diff
@@ -128,15 +128,7 @@
     (   Payload_Option = none % Last_Head_Id = Current_Head_Id
     ->  Result = same(Last_Head_Id)
     ;   Payload_Option = some(Payload),
-        catch(call(Push_Predicate, Remote_URL, Payload),
-              E,
-              (   E = error(Inner_E,_),
-                  (   Inner_E = history_diverged
-                  ;   Inner_E = remote_unknown
-                  ;   Inner_E = authorization_failure(_)
-                  ;   Inner_E = communication_failure(_))
-              ->  throw(error(remote_unpack_failed(Inner_E),_))
-              ;   throw(error(remote_unpack_unexpected_failure(E),_)))),
+        call(Push_Predicate, Remote_URL, Payload),
```

Comment:
how come we can do away with those catches?

Reply:
Because we catch the appropriate errors in `api_report_errors`. I don't see why we need to wrap these in `remote_unpack_failed` and `remote_unpack_unexpected_failure`. Thanks to the first argument `push` to `api_report_errors`, we know where the error is coming from. Also, `authorized_push` was simplified to wrap connection errors with `remote_connection_failure`, and `api_global_error_jsonld` now attempts to print nice error messages for `remote_connection_failure`.